test/realtime_web/controllers/channels_controller_test.exs

```diff
@@ -0,0 +1,55 @@
+defmodule RealtimeWeb.ChannelsControllerTest do
+  use RealtimeWeb.ConnCase, async: false
+
+  import Mock
+
+  alias Realtime.GenCounter
+  alias Realtime.Tenants
+
+  @cdc "postgres_cdc_rls"
+  @token "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1MTYyMzkwMjIsInJvbGUiOiJmb28iLCJleHAiOiJiYXIifQ.Ret2CevUozCsPhpgW2FMeFL7RooLgoOvfQzNpLBj5ak"
+
+  setup_with_mocks [
+                     {GenCounter, [], new: fn _ -> :ok end},
+                     {GenCounter, [], add: fn _ -> :ok end},
+                     {GenCounter, [], put: fn _, _ -> :ok end},
+                     {GenCounter, [], get: fn _ -> {:ok, 0} end}
+                   ],
+                   %{conn: conn} do
+    start_supervised(RealtimeWeb.Joken.CurrentTime.Mock)
+    tenant = tenant_fixture()
+    settings = Realtime.PostgresCdc.filter_settings(@cdc, tenant.extensions)
+    settings = Map.put(settings, "id", tenant.external_id)
+    settings = Map.put(settings, "db_socket_opts", [:inet])
+
+    start_supervised!({Tenants.Migrations, settings})
+    {:ok, db_conn} = Tenants.Connect.lookup_or_start_connection(tenant.external_id)
+
+    conn =
+      conn
+      |> put_req_header("accept", "application/json")
+      |> put_req_header("authorization", "Bearer #{@token}")
+      |> then(&%{&1 | host: "#{tenant.external_id}.supabase.com"})
+
+    on_exit(fn -> truncate_table(db_conn, "realtime.channels") end)
```

Comment:
Why is the table truncated? The sandbox is not available?

In any case, one tip is to truncate when the test starts, instead of `on_exit`, for two reasons:

1. If something fails spectacularly and you can't truncate, the next time you run the suite things will fail. If you truncate on setup, then it is always good
2. Sometimes you may want to inspect the table for debugging, but if you always truncate it after, you can't  :)

Reply:
unfortunately no since it's using Postgrex connections. I might be able to checkout a conn from Ecto but I had no clue on how to do it 😓